app/peoples/knights.py

```diff
@@ -0,0 +1,64 @@
+from __future__ import annotations
+from app.items.armour import Armour
+from app.items.weapon import Weapon
+from app.items.potions import Potion
+
+
+class Knight:
+    def __init__(self,
+                 name: str,
+                 power: int,
+                 hp: int,
+                 armour: list[Armour],
+                 weapon: Weapon,
+                 potion: Potion | None) -> None:
+        self.name = name
+        self.power = power
+        self.hp = hp
+        self.armour = armour
+        self.weapon = weapon
+        self.potion = potion
+        self.protection = 0
+        self.stats = {
+            "hp": self.hp,
+            "power": self.power,
+            "protection": self.protection
+        }
+
+    @classmethod
+    def from_dict(cls, knight_dict: dict) -> Knight:
+        if knight_dict["potion"]:
+            return cls(
+                knight_dict["name"],
+                knight_dict["power"],
+                knight_dict["hp"],
+                Armour.from_list_to_list(knight_dict["armour"]),
+                Weapon.from_dict(knight_dict["weapon"]),
+                Potion.from_dict(knight_dict["potion"])
+            )
+        return cls(
+            knight_dict["name"],
+            knight_dict["power"],
+            knight_dict["hp"],
+            Armour.from_list_to_list(knight_dict["armour"]),
+            Weapon.from_dict(knight_dict["weapon"]),
+            None
+        )
```

Comment:
make ur code DRY u can use get method for Potion.from_dict and make check for None insid from_dict method)